src/constructs/core/parameters.test.ts

```diff
@@ -18,6 +18,57 @@ import {
 } from "./parameters";
 import type { GuStack } from "./stack";
 
+describe("The GuParameter class", () => {
+  it("sets the type as passed through by default", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuParameter(stack, "Parameter", { type: "Boolean" });
+
+    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
+
+    expect(json.Parameters.Parameter).toEqual({
+      Type: "Boolean",
+    });
+  });
+
+  it("wraps the type with SSM utility is fromSSM is true", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuParameter(stack, "Parameter", { type: "Boolean", fromSSM: true });
+
+    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
+
+    expect(json.Parameters.Parameter).toEqual({
+      Type: "AWS::SSM::Parameter::Value<Boolean>",
+    });
+  });
+
+  it("defaults to string if SSM is true but no type provided", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuParameter(stack, "Parameter", { fromSSM: true });
+
+    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
+
+    expect(json.Parameters.Parameter).toEqual({
+      Type: "AWS::SSM::Parameter::Value<String>",
+    });
+  });
+
+  it("passes through other values without modification", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuParameter(stack, "Parameter", { type: "Boolean", fromSSM: true, description: "This is a test" });
```

Comment:
We might want to make `type` an enum or validate it at some point, for a later PR though.